integrations/redis.go

```diff
@@ -0,0 +1,83 @@
+package integrations
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-redis/redis/v7"
+
+	"github.com/reddit/baseplate.go/batcherror"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+// PipelineErrors is a set of errors from the commands that run in a Pipeline.
+type PipelineErrors []error
+
+// Error implements the error interface.
+func (v PipelineErrors) Error() string {
+	errorStrings := make([]string, len(v))
+	for i, err := range v {
+		errorStrings[i] = err.Error()
+	}
+	return strings.Join(errorStrings, "\n")
+}
+
+var _ error = (*PipelineErrors)(nil)
+
+// RedisSpanHook is a redis.Hook for wrapping Redis commands and pipelines
+// in Client Spans and metrics.
+type RedisSpanHook struct {
+	ClientName string
+}
+
+var _ redis.Hook = RedisSpanHook{}
+
+// BeforeProcess starts a client Span before processing a Redis command and
+// starts a timer to record how long the command took.
+func (h RedisSpanHook) BeforeProcess(ctx context.Context, cmd redis.Cmder) (context.Context, error) {
+	return h.startChildSpan(ctx, cmd.Name()), nil
+}
+
+// AfterProcess ends the client Span started by BeforeProcess, publishes the
+// time the Redis command took to complete, and a metric indicating whether the
+// command was a "success" or "fail"
+func (h RedisSpanHook) AfterProcess(ctx context.Context, cmd redis.Cmder) error {
+	return h.endChildSpan(ctx, cmd.Err())
+}
+
+// BeforeProcessPipeline starts a client span before processing a Redis pipeline
+// and starts a timer to record how long the pipeline took.
+func (h RedisSpanHook) BeforeProcessPipeline(ctx context.Context, cmds []redis.Cmder) (context.Context, error) {
+	return h.startChildSpan(ctx, "pipeline"), nil
+}
+
+// AfterProcessPipeline ends the client span started by BeforeProcessPipeline,
+// publishes the time the Redis pipeline took to complete, and a metric
+// indicating whether the pipeline was a "success" or "fail"
+func (h RedisSpanHook) AfterProcessPipeline(ctx context.Context, cmds []redis.Cmder) error {
+	var errs batcherror.BatchError
+	for _, cmd := range cmds {
+		if err := cmd.Err(); err != nil {
```

Comment:
nit: You don't even need the `!= nil` check with BatchError, you can just do `errs.Add(cmd.Err())` :)